cdap-storage-ext-spanner/src/main/java/io/cdap/cdap/storage/spanner/SpannerStructuredTableSchema.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright © 2025 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.storage.spanner;
+
+import io.cdap.cdap.spi.data.table.StructuredTableId;
+import io.cdap.cdap.spi.data.table.StructuredTableSchema;
+import io.cdap.cdap.spi.data.table.field.FieldType;
+import io.cdap.cdap.storage.spanner.compression.CompressionConfig;
+import io.cdap.cdap.storage.spanner.compression.CompressionConfig.CompressorType;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+
+public class SpannerStructuredTableSchema extends StructuredTableSchema {
+
+  /**
+   * Map of column name and compression config.
+   */
+  private final Map<String, CompressionConfig> compressionConfigMap;
+
+  SpannerStructuredTableSchema(StructuredTableId tableId,
+      List<FieldType> fields,
+      List<String> primaryKeys,
+      Collection<String> indexes,
+      Map<String, CompressionConfig> compressionConfigMap) {
+    super(tableId, fields, primaryKeys, indexes);
+    this.compressionConfigMap = compressionConfigMap;
```

Comment:
1. Make `this. compressionConfigMap` immutable / unmodifiable collection.

2. Even if `compressionConfigMap` is null, initialize the member variable to empty map.